cohortextractor2/concepts/tables.py

```diff
@@ -21,30 +22,40 @@ def __init__(self):
         super().__init__(Table("practice_registrations"))
 
 
-class Patients(TableContract):
+class PatientDemographics(TableContract):
     """
-    The Patients table holds personal details about the patient.
+    Provides demographic information about patients
     """
 
     patient_id = Column(
         type=types.PseudoPatientId(),
-        help=(
+        description=(
             "Patient's pseudonymous identifier, for linkage. You should not normally "
             "output or operate on this column"
         ),
+        help="",
+        constraints=[],
     )
     date_of_birth = Column(
         type=types.Date(),
-        help="The patient's date of birth",
+        description="Patient's year and month of birth.",
+        help="The day will always be the first of the month. Must be present.",
+        constraints=[DateConstraint(match_format=["%Y-%m-01"])],
```

Comment:
Where do we enforce required fields? Should we have a` RequiredConstraint` that checks there are no nulls in a column?

Reply:
Yes, I think that's another kind of constraint.